tests/utils/streams/test_kafka.py

```diff
@@ -17,268 +18,147 @@
+                **self.configuration,
+                "auto.offset.reset": auto_offset_reset,
                 "enable.auto.commit": "false",
                 "enable.auto.offset.store": "false",
                 "enable.partition.eof": "true",
-                "group.id": "test",
+                "group.id": group,
                 "session.timeout.ms": 10000,
             },
-            codec=codec,
+            self.codec,
         )
 
-    with closing(build_producer()) as producer:
-        value = uuid.uuid1().hex.encode("utf-8")
-        assert (
-            wait(
-                [producer.produce(topic, KafkaPayload(None, value)) for i in range(2)],
-                timeout=5.0,
-            ).not_done
-            == set()
+    def get_producer(self) -> KafkaProducer[int]:
+        return KafkaProducer(self.configuration, self.codec)
+
+    def test_auto_offset_reset_earliest(self) -> None:
+        with self.get_topic() as topic:
+            with closing(self.get_producer()) as producer:
+                producer.produce(topic, 0).result(5.0)
+
+            with closing(
+                self.get_consumer("test", auto_offset_reset="earliest")
+            ) as consumer:
+                consumer.subscribe([topic])
+
+                message = consumer.poll(10.0)
+                assert isinstance(message, Message)
+                assert message.offset == 0
+
+    def test_auto_offset_reset_latest(self) -> None:
+        with self.get_topic() as topic:
+            with closing(self.get_producer()) as producer:
+                producer.produce(topic, 0).result(5.0)
+
+            with closing(
+                self.get_consumer("test", auto_offset_reset="latest")
+            ) as consumer:
+                consumer.subscribe([topic])
+
+                try:
+                    consumer.poll(10.0)  # XXX: getting the subcription is slow
+                except EndOfPartition as error:
+                    assert error.partition == Partition(topic, 0)
+                    assert error.offset == 1
+                else:
+                    raise AssertionError("expected EndOfPartition error")
+
+    def test_auto_offset_reset_error(self) -> None:
+        with self.get_topic() as topic:
+            with closing(self.get_producer()) as producer:
+                producer.produce(topic, 0).result(5.0)
+
+            with closing(
+                self.get_consumer("test", auto_offset_reset="error")
+            ) as consumer:
+                consumer.subscribe([topic])
+
+                with pytest.raises(ConsumerError):
+                    consumer.poll(10.0)  # XXX: getting the subcription is slow
+
+    def test_commit_log_consumer(self) -> None:
+        # XXX: This would be better as an integration test (or at least a test
+        # against an abstract Producer interface) instead of against a test against
+        # a mock.
+        commit_log_producer = FakeConfluentKafkaProducer()
```

Comment:
This will be replaced with the new consumer soon, but that requires refactoring in the `ConsumerBuilder` which is a little daunting.